lua/nvim-lsp-setup/utils.lua

```diff
@@ -31,16 +31,16 @@ function M.default_mappings(bufnr, mappings)
 end
 
 function M.disable_formatting(client)
-    client.resolved_capabilities.document_formatting = false
-    client.resolved_capabilities.document_range_formatting = false
+    client.server_capabilities.document_formatting = false
+    client.server_capabilities.document_range_formatting = false
 end
 
 function M.format_on_save(client)
-    if client.resolved_capabilities.document_formatting then
+    if client.server_capabilities.document_formatting then
         vim.cmd([[
           augroup Format
             au! * <buffer>
-            au BufWritePre <buffer> lua vim.lsp.buf.formatting_sync(nil, 1000)
```

Comment:
Seems not compatible with nvim 0.7 ?

Reply:
,my neovim‘s version is 0.8, it works seems well.

Reply:
```
E5108: Error executing lua [string ":lua"]:1: attempt to call field 'format' 
```

We should be compatible with neovim 0.7

Reply:
thank you, I have changed it back.